src/Adapter/Title/Repository/TitleRepository.php

```diff
@@ -0,0 +1,141 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Title\Repository;
+
+use Gender;
+use PrestaShop\PrestaShop\Adapter\Title\Validate\TitleValidator;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\CannotAddTitleException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\CannotUpdateTitleException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Title\ValueObject\TitleId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+
+/**
+ * Provides access to Title data source
+ */
+class TitleRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var TitleValidator
+     */
+    private $titleValidator;
+
+    /**
+     * @param TitleValidator $titleValidator
+     */
+    public function __construct(
+        TitleValidator $titleValidator
+    ) {
+        $this->titleValidator = $titleValidator;
+    }
+
+    /**
+     * @param TitleId $titleId
+     *
+     * @return Gender
+     *
+     * @throws CoreException
+     * @throws TitleNotFoundException
+     */
+    public function get(TitleId $titleId): Gender
+    {
+        /** @var Gender $title */
+        $title = $this->getObjectModel(
+            $titleId->getValue(),
+            Gender::class,
+            TitleNotFoundException::class
+        );
+
+        return $title;
+    }
+
+    /**
+     * @param TitleId $titleId
+     *
+     * @throws CoreException
+     * @throws TitleNotFoundException
+     */
+    public function assertTitleExists(TitleId $titleId): void
+    {
+        $this->assertObjectModelExists(
+            $titleId->getValue(),
+            'gender',
+            TitleNotFoundException::class
+        );
+    }
+
+    /**
+     * @param Gender $title
+     * @param int $errorCode
+     *
+     * @return TitleId
+     */
+    public function add(Gender $title, int $errorCode = 0): TitleId
+    {
+        $this->titleValidator->validate($title);
+        $id = $this->addObjectModel(
+            $title,
+            CannotAddTitleException::class,
+            $errorCode
+        );
+
+        return new TitleId($id);
+    }
+
+    /**
+     * @param Gender $title
+     */
+    public function update(Gender $title): void
```

Comment:
is this method used? 
In update handler I see you use partialUpdate, so unless it is used somewhere, you can get rid of it

Reply:
Fixed